### PR TITLE
mingw-w64 9.0.0

### PR DESCRIPTION
**Pull request by Gcenx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/mingw-w64.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 class MingwW64 < Formula
   desc "Minimalist GNU for Windows and GCC cross-compilers"
   homepage "https://sourceforge.net/projects/mingw-w64/"
-  url "https://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v8.0.2.tar.bz2"
-  sha256 "f00cf50951867a356d3dc0dcc7a9a9b422972302e23d54a33fc05ee7f73eee4d"
+  url "https://downloads.sourceforge.net/project/mingw-w64/mingw-w64/mingw-w64-release/mingw-w64-v9.0.0.tar.bz2"
+  sha256 "1929b94b402f5ff4d7d37a9fe88daa9cc55515a6134805c104d1794ae22a4181"
   license "ZPL-2.1"
-  revision 1
+  revision 0
```
Comment (carlocab) [inline Formula/mingw-w64.rb:7]: Please omit this. We leave out `revision` whenever it's `0`.

File: Formula/mingw-w64.rb [hunk 2]
```diff
@@ -148,6 +148,18 @@ def install
...
+        --host=#{target}
+        --with-sysroot=#{arch_dir}/#{target}
+        --prefix=#{arch_dir}
+        --program-prefix=#{target}-
+      ]
+      mkdir "mingw-w64-tools/widl/build-#{arch}" do
+        system "../configure", *args
+        system "make"
+        system "make", "install"
+      end
```
Comment (carlocab) [inline Formula/mingw-w64.rb:161]: What does this build?

Comment (Gcenx) [reply Formula/mingw-w64.rb:161]: widl is a MIDL compiler.

```
An MIDL compiler processes the IDL file and creates a type library, a header, and proxy files.
```

Part of reason users really wanted a `wine` formula was to have this utility, mingw luckily now wine maintainers also work on mingw and keeping there headers/widl in sync with upstream wine.

File: Formula/mingw-w64.rb (hunk 1 above)
Comment (Gcenx) [reply Formula/mingw-w64.rb:7]: I’ll remove this and force push when I near my system.

File: Formula/mingw-w64.rb (hunk 2 above)
Comment (carlocab) [reply Formula/mingw-w64.rb:161]: Hm, ok. Can you double-check how much this adds to the installed bottle size?

Comment (Gcenx) [reply Formula/mingw-w64.rb:161]: widl is a lone binary each sized around 610KB, not sure about writing a test for this due to lack of familiarity with writing an idl file.

Comment (carlocab) [reply Formula/mingw-w64.rb:160]: Not even a minimal `"hello world"`-type test? It seems weird to try to enable functionality in a formula when we don't even know how to use it.

Comment (Gcenx) [reply Formula/mingw-w64.rb:160]: I've added a basic test that compiles `example.idl` into `example_s.c`, how would I test this file is created? (guessing I need to verify it was created)

Comment (carlocab) [reply Formula/mingw-w64.rb:160]: Something like
```ruby
assert_predicate testpath/"example_s.c", :exist, "example_s.c should have been created"
```
should work.

Comment (Gcenx) [reply Formula/mingw-w64.rb:160]: Done, rebased and pushed

**Review by carlocab — APPROVED** [review]

**Comment by Gcenx** [comment]
@carlocab would this require a revision bump if I’d added the patch from #78329 to get this to compile on Apple Silicon?

I’ve built and ran the tests for this on my Apple silicon Mac.

**Comment by carlocab** [comment]
I don't think that changes anything for the existing bottles, so a revision bump is not needed. We only bump the revision if we want users who already have it installed to get an updated version of the built bottles.

**Pull request closed without merging**